src/main/java/frc/robot/subsystems/Drive.java

```diff
@@ -209,6 +210,22 @@ public Command rotateToAngle(double targetAngle) {
         });
     }
 
+    public Command driveTillReadyToShoot() {
+        return runOnce(() -> {
+            aimAtSpeaker = true;
+        }).andThen(run(() -> {})).until(() -> {
+            double distanceToTarget = getRobotToTarget(getSpeakerTarget()).getNorm();
+            Logger.recordOutput("Distance to target", distanceToTarget);
+            return calculateRotateSpeedToTarget(() -> getSpeakerTarget()) < ANGLE_MAX_ERROR && distanceToTarget > 2.4 && distanceToTarget < 2.9;
```

Comment:
The 2.4 and 2.9 should be defined as `final double` at the top of the file so they're easier to tune.